### PR TITLE
NCN-9. GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by aclearskye**

### CODE REVIEW

File: __tests__/article_id-comments.test.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  it("should respond with a 404 error from error middleware", async () => {
+    const { body } = await request(app)
+      .get("/api/iAmBroken/1/whatsGoingOn")
+      .expect(404);
+  });
+  it("should respond with a 404 error from error middleware", async () => {
+    const { body } = await request(app)
+      .get("/api/neverGonna/giveYou/up")
+      .expect(404);
+  });
```
Comment (scar1377) [inline __tests__/article_id-comments.test.js:123]: Can replace these 404s with a 400 /api/articles/banana/comments

File: controller/article-comment-controller.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const { fetchCommentDataById } = require("../model/article-comment-model");
+
+exports.getCommentById = async (req, res, next) => {
+  try {
+    const id = req.params.article_id;
+    const fetchedCommentData = await fetchCommentDataById(id);
+    res.status(200).send(fetchedCommentData);
```
Comment (scar1377) [inline controller/article-comment-controller.js]: better to send an object with a key - send({comments: fetchedCommentData})

File: model/article-comment-model.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+exports.fetchCommentDataById = async (id) => {
+  validateID(id);
+
+  const results = await db.query(
+    "SELECT comment_id, votes, created_at, author, body FROM comments WHERE article_id = $1 ORDER BY comment_id ASC;",
+    [id]
+  );
+
+  validateRes(results);
```
Comment (scar1377) [inline model/article-comment-model.js]: What if you've got an article that doesn't have any comments? results.rows.length will also be 0, but it's not a 404 error, but a 200

**Review by scar1377 — COMMENTED** [review]
Hi Tommy. There's a situation needs to be considered - an article which doesn't have any comments related. With your code, this case will be handled as an error 404 not found. However, it is a 200, the comments array is returned successfully, but just with 0 comments in it

File: __tests__/article_id-comments.test.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+});
+describe("400 & 404's: Error handling w/multiple tests", () => {
+  it("should respond with a 404 error from error middleware", async () => {
+    const { body } = await request(app)
+      .get("/api/articles/1/rickAstley")
+      .expect(404);
+  });
+  it("should respond with a 404 error when given an ID that does not exist", async () => {
+    const { body } = await request(app)
+      .get("/api/articles/eatingBananas/comments")
```
Comment (scar1377) [inline __tests__/article_id-comments.test.js:120]: apparently your invalid id is better than mine lol

**Review by scar1377 — APPROVED** [review]
Good to merge

File: __tests__/article_id-comments.test.js (hunk 4 above)
Comment (aclearskye) [reply __tests__/article_id-comments.test.js:120]: oops!! 👀 👀  lmao

**Pull request merged**